src/g3log/logmessage.hpp

```diff
@@ -92,9 +98,9 @@ namespace g3 {
       LEVELS _level;
       std::string _expression; // only with content for CHECK(...) calls
       mutable std::string _message;
-
-
-
+#ifdef ENABLE_WIN_WSTRING_SUPPPORT
```

Comment:
The reason is that even in a windows program many log messages can and will be prepared as narrow strings. Dropping narrow string support would dramatically decrease performance in these cases as a utf8 -> utf16 conversion would have to be done before or during the logging request (i.e. in the execution thread).
Vice versa, the actual reason why anyone would want Utf-16 support in g3log is that converting log strings that are more quickly prepared in Utf-16 (anything involving strings coming from or going to the windows API) to Utf-8 before logging them will take its toll on the execution thread. 
Thus, from my point of view moving g2log to wchar_t only does not gain us anything. In this case it would be better to just prepare a layer that converts logging request from Utf-16 to Utf-8 in a convenient way.

My Idea was, on the other hand, to do all conversions in the logging thread and allowing sinks to deal with things like encoding and conversion.